### PR TITLE
Adds a paged Embed class.

### LINKED ISSUE #28: Create a class for embeds with pages
Embeds containing documentation on various symbols vary in length, and some symbols with a large amount of member types and functions contain too much information to contain in a single embed. The solution we had to this was to create an embed with "pages" which in reality is an embed whose content changes when certain reactions are received (i.e. right arrow reaction switches to the next "page" and left arrow switches back)

### PR DESCRIPTION
**Pull request by jchristgit**
This is a special type that allows users to navigate a set
of up to 20 embeds through reactions. Closes #28.

### CODE REVIEW

File: docflow/bot/util/paged_embed.py [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+    EMBED_EXPIRY
+, which sets the interval for
+an embed to "expire", in minutes.
+This is achieved through using
+    threading.Timer
+, which is started when
+the message is sent.
+"""
+
+from threading import Timer
```
Comment (emmiegit) [inline docflow/bot/util/paged_embed.py:26]: Why are you using `threading`?

File: docflow/bot/util/paged_embed.py [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+            ))
+            embed.add_page("🚗", EmbedPage(
+                title="Red!",
+                description="A red car. That's it!",
+                colour=discord.Colour.red()
+            ))
+            await embed.send()
+    """
+
+    def __init__(self, ctx, bot: commands.Bot, idx_emoji, *args, **kwargs):
```
Comment (emmiegit) [inline docflow/bot/util/paged_embed.py:122]: Why the `*args` and `**kwargs`?

File: docflow/bot/util/paged_embed.py [hunk 3]
```diff
@@ -0,0 +1,258 @@
...
+        on_reaction_add event which the
+        PagedEmbed uses to edit itself on a
+        relevant reaction. Normally, you do
+        not have to call this function
+        manually, it is called automatically
+        after a delay (see documentation for
+        this class itself.).
+        """
+
+        self._bot.remove_listener(self.on_reaction_add)
```
Comment (emmiegit) [inline docflow/bot/util/paged_embed.py:183]: `detach_listener` should be called in the object destructor

File: docflow/bot/util/paged_embed.py [hunk 4]
```diff
@@ -0,0 +1,258 @@
...
+        the sent message to show the users
+        the navigation choices.
+        """
+
+        self._msg = await self._ctx.send(embed=self.current_page)
+        for reaction in self._pages:
+            await self._msg.add_reaction(reaction)
+
+        # Start a timer to detach the event listener after the set interval
+        Timer(EMBED_EXPIRY * 60, self.detach_listener).start()
```
Comment (emmiegit) [inline docflow/bot/util/paged_embed.py:247]: Use `await asyncio.sleep()` or you're going to choke the bot

**Comment by emmiegit** [comment]
wtf I was reviewing it

File: docflow/bot/util/paged_embed.py (hunk 4 above)
Comment (jchristgit) [reply docflow/bot/util/paged_embed.py:247]: `threading.Timer()` starts a thread. I tested this, it doesn't block.

File: docflow/bot/util/paged_embed.py (hunk 1 above)
Comment (jchristgit) [reply docflow/bot/util/paged_embed.py:26]: See below.

File: docflow/bot/util/paged_embed.py (hunk 2 above)
Comment (jchristgit) [reply docflow/bot/util/paged_embed.py:122]: So it's possible to define an initial Embed as shown in the examples easily.

File: docflow/bot/util/paged_embed.py (hunk 3 above)
Comment (jchristgit) [reply docflow/bot/util/paged_embed.py:183]: You mean `__del__`?

**Pull request merged**